src/visualizations.jl

```diff
@@ -236,8 +260,100 @@ Scale given polygon by `s`, assuming polygon's center of reference is the origin
 function scale_polygon end
 export translate_polygon, scale_polygon, rotate_polygon
 
+############################################################################################
+## Visualization API
+############################################################################################
+
+"""
+    check_space_visualization_API(model::ABM)
+
+Checks whether all the necessary method extensions indicated in 
+[`space-visualization-API.jl`](../ext/AgentsVisualizations/space-visualization-API.jl) 
+have been defined.
+"""
+function check_space_visualization_API end
+
+## Required
+
+"""
+    agents_space_dimensionality(space::S) where {S<:Agents.AbstractSpace}
+
+Return dimensionality of given model space.
+"""
+function agents_space_dimensionality end
+
+"""
+    get_axis_limits!(model::ABM{S}) where {S<:Agents.AbstractSpace}
+
+Return appropriate axis limits for given model.
+Return `nothing, nothing` if you want to disable this.
+"""
+function get_axis_limits! end
+
+"""
+    agentsplot!(ax, model::ABM{S}, p::_ABMPlot) where {S<:Agents.AbstractSpace}
+
+Plot agents at their positions.
+"""
+function agentsplot! end
+
+## Preplots
+
+"""
+    spaceplot!(ax, model::ABM{S}; spaceplotkwargs...) where {S<:Agents.AbstractSpace}
+
+Create a space-dependent preplot.
```

Comment:
I think I have not understood why `agentsplot!` requires the `ABMPlot` object but `spaceplot!` does not.

Reply:
Because `agentsplot!` is dynamic which means that it updates over time whenever the model field inside the `ABMObservable` changes. `spaceplot!` on the other hand is static because (as far as I can anticipate right now) it only relies on the `space` field so we can simply pass `model` instead.

Reply:
If we want to unify the arguments of `agentsplot!` and `spaceplot!`, I would propose to use only `ABMPlot` because all the info is contained in there and remove the `model` argument. Would your prefer that or were you just confused about the differences in arguments taken by the two functions and wanted to clarify why?

Reply:
I think that having more similar arguments if possible is a good idea :+1: